src/main/java/com/example/onlinebookstore/service/order/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,147 @@
+            throw new EntityNotFoundException("There is no Order with id: " + id);
+        }
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getAllOrderItems(Long id, Pageable pageable) {
+        accessVerifyForOrderId(id);
+        return orderItemRepository.getOrderItemsByOrder(getOrderById(id), pageable)
+                .stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItemByOrderAndItemIds(Long orderId, Long itemId) {
+        accessVerifyForOrderId(orderId);
+        return getOrderById(orderId).getOrderItems()
+                .stream()
+                .filter(item -> item.getId().equals(itemId))
+                .findFirst()
+                .map(orderItemMapper::toDto)
+                .orElseThrow(() ->
+                        new EntityNotFoundException(
+                                "There is no item with id: " + itemId
+                                        + " for Order with id: " + orderId
+                        ));
+    }
+
+    private OrderResponseDto convertToDtoOrderAndSetItemsDto(Order order) {
+        OrderResponseDto response = orderMapper.toDto(order);
+        response.setOrderItems(order.getOrderItems()
+                .stream()
+                .map(orderItemMapper::toDto)
+                .collect(Collectors.toSet())
+        );
+        return response;
+    }
+
+    private BigDecimal multiplyQuantityByPrice(OrderItem item) {
+        BigDecimal price = item.getPrice();
+        int quantity = item.getQuantity();
+        return price.multiply(BigDecimal.valueOf(quantity));
+    }
+
+    private BigDecimal getTotal(Order order) {
+        BigDecimal total = BigDecimal.ZERO;
+        for (OrderItem item : order.getOrderItems()) {
+            total = total.add(multiplyQuantityByPrice(item));
+        }
+        return total;
+    }
+
+    private ShoppingCart getShoppingCartByUser(User user) {
+        return shoppingCartRepository.findByUser(userService.getCurrentUser())
+                .orElseThrow(() ->
+                        new EntityNotFoundException(
+                                "There is no Shopping Cart for User: "
+                                        + userService.getCurrentUser()
+                        ));
+    }
+
+    private Order getOrderById(Long id) {
+        return orderRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("There is no Order with id: " + id));
+    }
+
+    private void accessVerifyForOrderId(Long id) {
+        List<Long> availableIds = orderRepository.getOrdersByUser(userService.getCurrentUser())
+                .stream()
+                .map(Order::getId)
+                .toList();
+        if (!availableIds.contains(id)) {
+            throw new AccessDeniedException(
+                    "Your are not allowed to see Items of Order with id: " + id
+            );
+        }
```

Comment:
Create a helper method for that:
```
          boolean exists = orderRepository.existsByIdAndUser(id, userService.getCurrentUser());
        if (!exists) {
            throw new AccessDeniedException("Your are not allowed to see Items of Order with id: " + id);
        }
```
        in repo:
            boolean existsByIdAndUser(Long id, User user);